gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -228,7 +228,10 @@ protected void processMessage(DecodeableKafkaRecord<byte[],byte[]> message) {
           String tableName = jobStatusTableName(flowExecutionId, jobGroup, jobName);
           String status = jobStatus.getProp(JobStatusRetriever.EVENT_NAME_FIELD);
 
-          if (updatedJobStatus.getRight() == NewState.FINISHED) {
+          boolean retryRequired = modifyStateIfRetryRequired(jobStatus);
```

Comment:
call to modifyStateIfRetryRequired is pushed later in time (in code diff it is being shown above the older position, but actually that code runs before this one.